include/unifex/let_value_with_stop_token.hpp

```diff
@@ -194,13 +202,17 @@ struct _stop_token_operation<
         is_nothrow_connectable_v<inner_sender_t, receiver_t>;
 
   private:
+    template <typename Receiver2>
     auto connect_inner_op(
         SuccessorFactory& func,
         inplace_stop_token st,
-        Receiver&& r) noexcept(successor_is_nothrow&&
-                                   inner_receiver_nothrow_constructible&&
-                                       nothrow_connectable) {
-      return unifex::connect(func(st), receiver_t{this, st, (Receiver &&) r});
+        // we need to take r by reference to avoid problems at the call site
+        // related to unsequenced function argument evaluation
+        Receiver2&& r) noexcept(successor_is_nothrow&&
+                                    inner_receiver_nothrow_constructible&&
+                                        nothrow_connectable) {
```

Comment:
Is `std::is_nothrow_constructible_v<Receiver, Receiver2>` not needed here?

Reply:
`inner_receiver_nothrow_constructible` computes that for the old cold; I'll need to update it.